todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt

```diff
@@ -56,19 +56,20 @@ class TasksLocalDataSource private constructor(context: Context) : TasksDataSour
                 }
                 tasks.add(task)
             }
-            if (tasks.isNotEmpty()) {
-                callback.onTasksLoaded(tasks)
-            } else {
-                // This will be called if the table is new or just empty.
-                callback.onDataNotAvailable()
-            }
             close()
         }
         db.close()
+
+        if (!tasks.isEmpty()) {
```

Comment:
@DougSig I think that main reason of this kind of code extracting is that both of callbacks are not strict correlated with database, but with `ArrayList<Task> tasks`. I suppose that an author of commit wanted first copy values from database to tasks, then check if this `List` is empty.

And yes, you're right. Expression` !tasks.isEmpty()` looks a bit weird, restoring previous `tasks.isNotEmpty()` would cause much readable code ;-)